assets/terraform-modules/aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl

```diff
@@ -55,32 +52,30 @@ systemd:
         [Service]
         ConditionPathExists=/etc/kubernetes/kubeconfig
         EnvironmentFile=/etc/kubernetes/kubelet.env
-        Environment="RKT_RUN_ARGS=--uuid-file-save=/var/cache/kubelet-pod.uuid \
-          --volume=resolv,kind=host,source=/etc/resolv.conf \
-          --mount volume=resolv,target=/etc/resolv.conf \
-          --volume var-lib-cni,kind=host,source=/var/lib/cni \
-          --mount volume=var-lib-cni,target=/var/lib/cni \
-          --volume var-lib-calico,kind=host,source=/var/lib/calico \
-          --mount volume=var-lib-calico,target=/var/lib/calico \
-          --volume opt-cni-bin,kind=host,source=/opt/cni/bin \
-          --mount volume=opt-cni-bin,target=/opt/cni/bin \
-          --volume var-log,kind=host,source=/var/log \
-          --mount volume=var-log,target=/var/log \
-          --volume etc-cni-netd,kind=host,source=/etc/cni/net.d \
-          --mount volume=etc-cni-netd,target=/etc/cni/net.d \
-          --insecure-options=image"
-        ExecStartPre=/bin/mkdir -p /opt/cni/bin
-        ExecStartPre=/bin/mkdir -p /etc/kubernetes/manifests
-        ExecStartPre=/bin/mkdir -p /etc/cni/net.d
-        ExecStartPre=/bin/mkdir -p /etc/kubernetes/checkpoint-secrets
-        ExecStartPre=/bin/mkdir -p /etc/kubernetes/inactive-manifests
-        ExecStartPre=/bin/mkdir -p /var/lib/cni
-        ExecStartPre=/bin/mkdir -p /var/lib/calico
         ExecStartPre=/bin/mkdir -p /var/lib/kubelet/volumeplugins
         ExecStartPre=/usr/bin/bash -c "grep 'certificate-authority-data' /etc/kubernetes/kubeconfig | awk '{print $2}' | base64 -d > /etc/kubernetes/ca.crt"
         ExecStartPre=/etc/kubernetes/configure-kubelet-cgroup-driver
-        ExecStartPre=-/usr/bin/rkt rm --uuid-file=/var/cache/kubelet-pod.uuid
-        ExecStart=/usr/lib/coreos/kubelet-wrapper \
+        ExecStart=docker run \
+          --network=host \
+          --pid=host \
+          --privileged \
+          -v /dev:/dev:rw \
+          -v /etc/cni/net.d:/etc/cni/net.d:rw \
+          -v /etc/kubernetes:/etc/kubernetes:ro \
+          -v /etc/machine-id:/etc/machine-id \
+          -v /lib/modules:/lib/modules \
+          -v /run:/run:rw \
+          -v /sys:/sys:rw \
+          -v /opt/cni/bin:/opt/cni/bin:ro \
+          -v /usr/lib/os-release:/etc/os-release:ro \
+          -v /usr/sbin/iscsiadm:/usr/sbin/iscsiadm:rw \
+          -v /var/lib/calico:/var/lib/calico:ro \
+          -v /var/lib/cni:/var/lib/cni:rw \
+          -v /var/lib/docker:/var/lib/docker:rw \
```

Comment:
Not sure I follow why is this needed

Reply:
Because kubelet checks if the images are pulled in this directory IIRC and creates symlinks from Docker logs to pod logs:
```
      // Required by kubelet when creating Docker containers. This is required, when using Docker as container
      // runtime, as cAdvisor, which is integrated into kubelet will try to identify image read-write layer for
      // container when creating a handler for monitoring. This is needed to report disk usage inside the container.
      //
      // It is also needed, as kubelet creates a symlink from Docker container's log file to /var/log/pods.
```

Reply:
@invidian thanks! Where does that comment come from? Would it make sense to add it here? We have limited space for ignition config files, though :-/

Reply:
https://github.com/flexkube/libflexkube/blob/f6d4192504a1f373bac7902cb7b3bf180148dfbd/pkg/kubelet/kubelet.go#L364-L370 :wink:

Reply:
haha, nice. But we need to state this in lokomotive somehow (a comment here will eat our user_data size :-/)

Reply:
`a comment here will eat our user_data size` I don't think so. Aren't comments stripped from Ignition config when being transpiled by `ct`?

Reply:
To be honest, I'm not sure. I think comments are not stripped for things like inline scripts, but might be removed for this part.

Reply:
YAML comments should be stripped. Inline comments in scripts, are not obviously.

Reply:
Exactly... I guess this is not a yaml comment, as it is inside a command to run. That was my point.

Reply:
So what is that you folks are suggesting to do here?

Reply:
We could document that in self-hosted kubelet chart maybe? As the configuration between the two should always be in sync.

Reply:
Added it in the self hosted kubelet, wherever necessary.